service/src/test/java/bio/terra/profile/service/azure/AzureServiceUnitTest.java

```diff
@@ -105,48 +107,130 @@ public void getManagedApps() {
+            profileDao);
 
-    var result = azureService.getAuthorizedManagedAppDeployments(subId, user);
+    var result = azureService.getAuthorizedManagedAppDeployments(subId, true, user);
 
     assertEquals(result.size(), 1, "Duplicate app instances should be removed");
   }
+
+  @Test
+  public void getManagedApps_includeAssignedApplications() {
+    var applicationName = "fake_app_1";
+    var assignedTerraAppManagedResourceGroupId = "assigned_fake_mrg";
+    var unassignedTerraAppManagedResourceGroupId = "unassigned_fake_mrg";
+
+    var crlService = mock(CrlService.class);
+    var profileDao = mock(ProfileDao.class);
+    when(profileDao.listManagedResourceGroupsInSubscription(tenantId, subId))
+        .thenReturn(List.of(assignedTerraAppManagedResourceGroupId));
+
+    var assignedTerraApp = mock(Application.class);
+    mockApplicationCalls(assignedTerraApp,
+        offerName,
+        offerPublisher,
+        Optional.of(authedUserEmail),
+        assignedTerraAppManagedResourceGroupId,
+        applicationName);
+
+    var unassignedTerraApp = mock(Application.class);
+    mockApplicationCalls(unassignedTerraApp,
+        offerName,
+        offerPublisher,
+        Optional.of(authedUserEmail),
+        unassignedTerraAppManagedResourceGroupId,
+        applicationName);
+
+    var appsList =
+        Stream.of(assignedTerraApp, unassignedTerraApp);
+    var appService = mock(ApplicationService.class);
+    when(appService.getApplicationsForSubscription(eq(subId))).thenReturn(appsList);
+    when(appService.getTenantForSubscription(subId)).thenReturn(tenantId);
+
+    var offer = new AzureConfiguration.AzureApplicationOffer();
+    offer.setName(offerName);
+    offer.setPublisher(offerPublisher);
+    offer.setAuthorizedUserKey(authorizedUserKey);
+    var offers = Set.of(offer);
+    var azureService =
+        new AzureService(
+            crlService,
+            appService,
+            new AzureConfiguration("fake", "fake", "fake", offers, ImmutableSet.of()),
+            profileDao);
+
+    AzureManagedAppModel assignedAzureManagedAppModel = new AzureManagedAppModel()
+        .tenantId(tenantId)
+        .subscriptionId(subId)
+        .managedResourceGroupId(assignedTerraAppManagedResourceGroupId)
+        .applicationDeploymentName(applicationName);
+
+    AzureManagedAppModel unassignedAzureManagedAppModel = new AzureManagedAppModel()
+        .tenantId(tenantId)
+        .subscriptionId(subId)
+        .managedResourceGroupId(unassignedTerraAppManagedResourceGroupId)
+        .applicationDeploymentName(applicationName);
+
+    var includeAssignedResult = azureService.getAuthorizedManagedAppDeployments(subId, true, user);
+    assertEquals(List.of(assignedAzureManagedAppModel, unassignedAzureManagedAppModel),
+        includeAssignedResult);
+
+    appsList = Stream.of(assignedTerraApp, unassignedTerraApp);
+    when(appService.getApplicationsForSubscription(eq(subId))).thenReturn(appsList);
```

Comment:
I had to introduce this to get around Streams only being readable one time. When I call `getAuthorizedManagedAppDeployments` for the second time below, the mock reuses the same `appsList` Stream it used the first time the method was called. However, that Stream has already been read, so it throws an exception. 

I don't like it, but I couldn't figure out how to get Streams to do what I wanted so this was my workaround. If anyone knows of a better way, let me know!